esmvaltool/diag_scripts/land_carbon_cycle/diag_global_turnover.py

```diff
@@ -0,0 +1,798 @@
+
+    # save and close figure
+    t_x = plt.figtext(0.5, 0.5, ' ', transform=plt.gca().transAxes)
+    plut.save_figure(plot_path, _extr_art=[t_x])
+    plt.close()
+
+
+def main(diag_config):
+    """
+    Evaluate global distribution of ecosystem carbon turnover time.
+
+    Argument:
+    --------
+        diag_config - nested dictionary of metadata
+    """
+    my_files_dict = group_metadata(diag_config['input_data'].values(),
+                                   'dataset')
+
+    # get the data from the observation
+    global_tau_obs = _get_obs_data(diag_config)
+    base_name = ('{title}_{source_label}_'
+                 '{grid_label}'.format(
+                     title=global_tau_obs['grid']['tau_ctotal'].long_name,
+                     source_label=diag_config['obs_info']['source_label'],
+                     grid_label=diag_config['obs_info']['grid_label']))
+
+    global_tau_mod = {}
+    global_tau_mod['grid'] = {}
+    global_tau_mod['global'] = {}
+    provenance_record = get_provenance_record(
+        "Comparison of global distributions of turnover time of carbon"
+        "with matrix comparison, maps, and multimodel bias. Reproduces"
+        "figure 3 in Carvalhais et al. (2014).", ['mean', 'perc'], ['global'],
+        _get_ancestor_files(diag_config, 'tau_ctotal'))
+
+    with ProvenanceLogger(diag_config) as provenance_logger:
+        for key, value in my_files_dict.items():
+            global_tau_mod[key] = {}
+
+            # load the data
+            ctotal = _load_variable(value, 'ctotal')
+            gpp = _load_variable(value, 'gpp')
+            tau_ctotal = _calc_turnover(ctotal, gpp, key)
+            global_tau_mod['grid'][key] = tau_ctotal
+
+            # apply the GPP threshold and set the data in dictionary
+            gpp_global = gpp.collapsed(['latitude', 'longitude'],
+                                       iris.analysis.SUM)
+            ctotal_global = ctotal.collapsed(['latitude', 'longitude'],
+                                             iris.analysis.SUM)
+            tau_global = ctotal_global / gpp_global
+            tau_global.convert_units('yr')
+
+            global_tau_mod['global'][key] = np.float(tau_global.core_data())
+
+            if diag_config['write_plots']:
+                base_name_mod = (
+                    'global_{title}_{source_label}_'
+                    '{grid_label}'.format(
+                        title=global_tau_obs['grid']['tau_ctotal'].long_name,
+                        source_label=key,
+                        grid_label=diag_config['obs_info']['grid_label']))
+                plot_path_mod = get_plot_filename(base_name_mod, diag_config)
+                # plot_path_list.append(plot_path_mod)
+                _plot_single_map(plot_path_mod, tau_ctotal,
+                                 global_tau_mod['global'][key], key,
+                                 diag_config)
+                provenance_logger.log(plot_path_mod, provenance_record)
```

Comment:
It looks like this is a plot for a single model, so I would expect the ancestors to be just the files corresponding to this model, or did I not follow your code correctly?

Reply:
Yes. You are correct.

Reply:
In that case, why are you using the same provenance record for all plots?